src/pages/RepoPage/BundlesTab/BundleContent/BundleChart/useBundleChartData.ts

```diff
@@ -0,0 +1,171 @@
+    dayDiff = differenceInCalendarDays(today, after)
+  }
+
+  let interval: TimeseriesIntervals = TimeseriesInterval.INTERVAL_7_DAY
+  if (dayDiff < DAY_THRESHOLD) {
+    interval = TimeseriesInterval.INTERVAL_1_DAY
+  } else if (dayDiff > MONTH_THRESHOLD) {
+    interval = TimeseriesInterval.INTERVAL_30_DAY
+  }
+
+  return {
+    interval,
+    after: after,
+  }
+}
+
+interface UseBundleChartArgs {
+  provider: string
+  owner: string
+  repo: string
+  branch: string
+  bundle: string
+}
+
+export function useBundleChartData({
+  provider,
+  owner,
+  repo,
+  branch,
+  bundle,
+}: UseBundleChartArgs) {
+  const { params } = useLocationParams()
+  const { data: overview } = useRepoOverview({ provider, owner, repo })
+
+  // @ts-expect-error - useLocationParams needs fixing
+  const trend = params?.trend ?? Trend.THREE_MONTHS
+  const today = useMemo(() => new Date(), [])
+
+  const queryVars = useMemo(
+    () =>
+      createQueryVars({
+        today,
+        trend,
+        oldestCommitAt: overview?.oldestCommitAt ?? null,
+      }),
+    [overview?.oldestCommitAt, today, trend]
+  )
+
+  const { data: trendData, isLoading } = useBundleTrendData({
+    provider,
+    owner,
+    repo,
+    branch,
+    bundle,
+    interval: queryVars.interval,
+    after: queryVars.after,
+    before: today,
+    // this will be replaced once we have filtering by types implemented
+    filters: {
+      assetTypes: ['REPORT_SIZE'],
+    },
+    enabled: !!overview?.oldestCommitAt,
+    suspense: false,
+  })
+
+  const mergedData = useMemo(() => {
+    // merge the data from all the measurements into a single data point
+    // this will be required when we are able to filter by multiple asset types
+    // and we want to be able to merge their sizes into a single data point
+    // using a map for this because it's efficient for setting and getting values
+    const mergedDataMap = new Map<string, number>()
+    for (const x of trendData ?? []) {
+      let prevSize = 0
+      for (const y of x.measurements ?? []) {
+        const size = y.avg ?? prevSize
+        const presentEntry = mergedDataMap.get(y.timestamp)
+
+        if (prevSize !== size) {
+          prevSize = size
+        }
+
+        if (presentEntry) {
+          mergedDataMap.set(y.timestamp, presentEntry + size)
+        } else {
+          mergedDataMap.set(y.timestamp, size)
+        }
+      }
+    }
+
+    // take the merged data and convert it into an array of objects
+    return Array.from(mergedDataMap).map(([timestamp, avg]) => ({
+      date: new Date(timestamp),
+      size: avg,
+    }))
+  }, [trendData])
+
+  // find the max size to calculate the y-axis max value
+  let maxSize = 0
+  mergedData.forEach((x) => {
+    if (x.size > maxSize) {
+      maxSize = x.size
+    }
+  })
+  const multiplier = findBundleMultiplier(maxSize)
+
+  // calculate the y-axis max value based on the max size
+  // basically 2 + the next 10^x number, eg maxSize = 1000, maxY = 3000
```

Comment:
How was this decided on? If I'm understanding correctly, this seems unnecessarily large? Doesn't this mean all data points are in the bottom third of the graph?

Reply:
Whoops sorry my mental math is wrong, if the size is 1000, the output of this calculation will be 2